krabs/krabs/trace.hpp

```diff
@@ -212,6 +212,7 @@ namespace krabs {
          */
         size_t buffers_processed() const;
 
+
```

Comment:
Hmm, let's pass a `const trace_context &` instead of the schema_locator.
That way, if we need to thread other things through, we'll just add them to the context.
```suggestion
struct trace_context
{
    const schema_locator schema_locator_;
    /* Add additional trace context here. */
};
```

This will allow per-trace objects to be passed along on the callback which may be useful in the future if you have a single callback registered with multiple traces.